### PR TITLE
chore: use vscode-prettier to format; cleanup settings

### PR DESCRIPTION
**Pull request by palpatim**
#### Description of changes

This is a minor quality-of-life improvement aimed directly at my quality of life.

We currently specify `dbaeumer.vscode-eslint` as the project's preferred formatter, but that plugin has [known issues with the **Format Document** command](https://github.com/microsoft/vscode-eslint/issues/1355). More importantly, we use `prettier` for our CI lint checks.

This PR suggests the `esbenp.prettier-vscode` plugin and configures it as the default formatter for TS & JS. I also removed some unknown or deprecated settings from the settings.json file.

As a final pedantic, overly-nitpicky cherry on this pedantry sundae, I sorted the `settings.json` file and the `scripts` stanza of `package.json`, because it was a pain in the tuchus trying to find a particular script.

**UPDATE** Now with more E2E test region randomization! [23cfaeb](https://github.com/aws-amplify/amplify-category-api/pull/2051/commits/23cfaeb638c83c2a7ec481614f288c3c970933a2) removes RDS E2E tests from the region override map.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,28 +1,24 @@
...
+  "eslint.validate": ["javascript", "javascriptreact", "typescript", "typescriptreact"],
   "files.exclude": {
     "**/.git": true,
     "**/node_modules": true,
     "**/build": true,
     "**/coverage": true
   },
-  "editor.formatOnSave": false,
-  "eslint.validate": ["javascript", "javascriptreact", "typescript", "typescriptreact"],
-  "eslint.alwaysShowStatus": true,
```
Comment (palpatim) [inline .vscode/settings.json:11]: Removed: unknown option

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,28 +1,24 @@
...
-  "eslint.format.enable": true,
-  "eslint.lintTask.enable": true,
-  "eslint.packageManager": "yarn",
```
Comment (palpatim) [inline .vscode/settings.json:14]: Removed: deprecated option

File: .vscode/settings.json [hunk 3]
```diff
@@ -1,28 +1,24 @@
...
   "files.exclude": {
     "**/.git": true,
     "**/node_modules": true,
     "**/build": true,
     "**/coverage": true
   },
-  "editor.formatOnSave": false,
-  "eslint.validate": ["javascript", "javascriptreact", "typescript", "typescriptreact"],
-  "eslint.alwaysShowStatus": true,
-  "eslint.format.enable": true,
```
Comment (palpatim) [inline .vscode/settings.json:12]: Retaining this to support team members who still use eslint to format on save.

**Review by palpatim — COMMENTED** [review]
marker review

**Review by alharris-at — APPROVED** [review]

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**